Improvements in subscription site list

JuanLucha opened the pull request:
Fixes https://github.com/Automattic/wp-calypso/issues/74966

## Proposed Changes

This pull request addresses several improvements to the way sites are displayed. The following points have been fixed:

1. **Handle a site with no logo**: Sites that don't have a logo will now be displayed correctly, with a default globe icon (like in Reader).
2. **Display site logos with a circular mask applied**: Site logos will now be displayed with a circular mask, giving a consistent and visually appealing appearance.
3. **Display the date with the expected date format applied**: The site's date will now be displayed using the expected date format, improving readability and consistency across the application. **Note**: the format is using en-US for now. When we can access real data, we will use the user locale instead.
4. **Handle long text**: Long site names and other text elements are handled gracefully, ensuring the layout remains clean and uncluttered. We are using ellipsis for this.
5. **Handle site logo, site name, and site URL click**: Clicking on the site logo, site name, or site URL will trigger the appropriate event or action, providing a better user experience.
6. **Only the site's domain name should be displayed**: To improve readability and focus on the essential information, only the site's domain name will be displayed rather than the full URL.

<img width="1053" alt="Screenshot 2023-03-30 at 12 45 29" src="https://user-images.githubusercontent.com/3832570/228812529-5901adb9-bbb6-451f-951f-e30204196fcd.png">

## Testing Instructions

- Apply these changes.
- Please check the 6 points described above, making sure the experience is now closer to the Figma design.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React ,or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

JuanLucha committed:
Improvements in subscription site list

ivan-ottinger commented on client/landing/subscriptions/site-list/site-row.tsx:
```diff
@@ -1,17 +1,41 @@
 export default function SiteRow( { id, name, icon, url, date, emailFrequency }: SiteType ) {
+	const hostname = useMemo( () => new URL( url ).hostname, [ url ] );
+
+	// For now, using `en-US` locale, but we should use the user's locale when we receive real data for the component
```
I am afraid the current API implementation doesn't provide the user's locale. 🤔 

Looking at the current Reskinned / old portal, the date is not translated there at all:
![Markup on 2023-03-30 at 13:43:26](https://user-images.githubusercontent.com/25105483/228825561-a7b973ce-b057-4749-a0d6-34e80cb526ea.png)

I think we could address this issue in a separate task - as the change will likely need more work.

ivan-ottinger commented on client/landing/subscriptions/site-list/site-row.tsx:
```diff
@@ -1,17 +1,41 @@
-				<span className="title-column">
-					<span className="name">{ name }</span>
-					<span className="url">{ url }</span>
+			<a href={ url } className="title-box">
```
I wonder if it would make sense to open the link in a new tab. Maybe it is better to keep it as it is now - since the user can still make the link open in a new tab if they want.

JuanLucha replied:
Indeed. Opening in a new tab now 👍

TimBroddin replied:
We decided not to open in new tabs. Context: p1680186817428589-slack-C02TCEHP3HA

ivan-ottinger reviewed (commented):
I haven't tested the PR yet, but wanted to share two thought that came to my mind while I was reviewing the code for the first time.

TimBroddin committed:
Update site-row.tsx

Added rel="noreferrer noopener"

Noreferrer to not leak referrer.
Noopener to avoid access to `window.opener` when opened in new tab .

TimBroddin commented on client/landing/subscriptions/site-list/site-row.tsx:
```diff
@@ -1,25 +1,41 @@
-				<span className="title-column">
-					<span className="name">{ name }</span>
-					<span className="url">{ URL }</span>
+			<a href={ url } rel="noreferrer noopener" className="title-box">
```
We decided not to open in new tabs. Context: p1680186817428589-slack-C02TCEHP3HA

I added rel="noreferrer noopener" here, though.

TimBroddin replied:
This is nitpicking, but `class "title-box"` is being applied to both the link and the span. This confused me a bit semantically.

Is this needed? 

From my limited testing, it seems like you could replace the class of the span with a new class, something like `.title-block` and change line 27 to:`.title-box, .title-block { `

Really nitpicking though, but reading the code I thought you made a mistake at first 😀

JuanLucha replied:
That is a perfect point. Thanks for these detailed comments, @TimBroddin 👍

TimBroddin reviewed (approved):
I made a change, and made a nitpicking comment ☺️

One thing I noticed is that your branch name `more-details-in-site-list` is not how it's usally done in our repositories. Most people prefix branches with `add/`, `update/` or `fix/`. Just a small suggestion to keep everything organized.
